src/gui/cli/ClientApi.cpp

```diff
@@ -113,30 +115,51 @@ namespace Zappy::GUI {
         static std::unordered_map<std::string, std::function<void(ClientApi &, std::string)>> myResponses = {
             {"WELCOME", &ClientApi::ReceiveWelcome},
             {"msz", &ClientApi::ReceiveMsz},
+            {"bct", &ClientApi::ReceiveBct},
         };
 
-        while (_readBuffer.find("\n") != std::string::npos) {
-            std::string myResponse = _readBuffer.substr(0, _readBuffer.find("\n"));
-            std::string myCommand = myResponse.substr(0, myResponse.find(" "));
-            std::string myArgs = myResponse.substr(myResponse.find(" ") + 1);
+        while (_readBuffer.find('\n') != std::string::npos) {
+            std::string const myResponse = _readBuffer.substr(0, _readBuffer.find('\n'));
+            std::string const myCommand = myResponse.substr(0, myResponse.find(' '));
+            std::string const myArgs = myResponse.substr(myResponse.find(' ') + 1);
 
             if (myResponses.find(myCommand) != myResponses.end()) {
                 myResponses[myCommand](*this, myArgs);
             }
-            _readBuffer = _readBuffer.substr(_readBuffer.find("\n") + 1);
+            _readBuffer = _readBuffer.substr(_readBuffer.find('\n') + 1);
         }
     }
 
-    void ClientApi::ReceiveWelcome(__attribute__((unused)) std::string aResponse)
+    void ClientApi::ReceiveWelcome(__attribute__((unused)) const std::string &aResponse)
     {
         _writeBuffer += _teamName + "\n";
     }
 
-    void ClientApi::ReceiveMsz(std::string aResponse)
+    void ClientApi::ReceiveMsz(const std::string &aResponse)
     {
-        std::string myX = aResponse.substr(0, aResponse.find(" "));
-        std::string myY = aResponse.substr(aResponse.find(" ") + 1);
+        std::string const myX = aResponse.substr(0, aResponse.find(' '));
+        std::string const myY = aResponse.substr(aResponse.find(' ') + 1);
 
         _serverData._mapSize = std::make_pair(std::stoi(myX), std::stoi(myY));
     }
+
+    void ClientApi::ReceiveBct(const std::string &aResponse)
```

Comment:
i need to look with @Valegox why my function is called 20 times at welcome (i think it's because he see the command ```bct``` at welcome